test/runnable/test16640.d

```diff
@@ -0,0 +1,11 @@
+// PERMUTE_ARGS:
+
+void testFileFullPathAsDefaultArgument(string preBakedFileFullPath, string fileFullPath = __FILE_FULL_PATH__)
+{
+    assert(preBakedFileFullPath == fileFullPath);
+}
+
+void main()
+{
+    testFileFullPathAsDefaultArgument(__FILE_FULL_PATH__);
```

Comment:
There already is a test in test/compilable/line.d that would fail if that would be the case.